File: daily/Linked List/2-Add-two-numbers.py
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        dummy = ListNode(0)
        curr = dummy
        curry = 0
        while l1 or l2 or curry > 0:
            if l1:
                curry += l1.val
                l1 = l1.next
            if l2:
                curry += l2.val
                l2 = l2.next

            curr.next = ListNode(curry % 10)
            curr = curr.next
            curry //= 10
        return dummy.next

File: daily/Linked List/2-Add-two-numbers.go
package daily

func addTwoNumbers(l1 *ListNode, l2 *ListNode) *ListNode {
	dummy := &ListNode{0, nil}
	curr := dummy
	carry := 0
	for l1 != nil || l2 != nil || carry > 0 {
		if l1 != nil {
			carry += l1.Val
			l1 = l1.Next
		}
		if l2 != nil {
			carry += l2.Val
			l2 = l2.Next
		}
		curr.Next = &ListNode{carry % 10, nil}
		curr = curr.Next
		carry /= 10
	}
	return dummy.Next
}
